internal/plugin/plugin.go

```diff
@@ -157,13 +157,18 @@ func (m *Manager) createFile(
 		urlForInput = pkg.URLForFlakeInput()
 	}
 
+	packages, err := m.InstallablePackageNames()
```

Comment:
I think it's fine for this to be all package names. Specifically, this is used so that plugin templates know all packages, it is currently used by php and haskell plugins.

If anything, in a follow up we could change this to be Package structs instead of strings. (and it would be backward compatible because the struct has a `String()` method.

This would allow us to remove InstallablePackageNames and just keep a single `PackageNames`.

Reply:
> If anything, in a follow up we could change this to be Package structs instead of strings. (and it would be backward compatible because the struct has a String() method.

Before we rely on `.String()`, I'd prefer making Package structs not embed url.URL, so we can properly implement `String()` to do what is intended.